### PR TITLE
feat: expose `keepFnNames` and `keepClassNames` options of builtin swc minfier

### PR DESCRIPTION
**Pull request by xinxinhe1810**
## Summary

feat for https://github.com/web-infra-dev/rspack/issues/4064

## Test Plan

Added

### CODE REVIEW

File: crates/rspack_binding_options/src/options/raw_builtins/raw_swc_js_minimizer.rs [hunk 1]
```diff
@@ -30,6 +30,8 @@ pub struct RawMinificationConditions {
 pub struct RawMinification {
   pub passes: u32,
   pub drop_console: bool,
+  pub keep_classnames: bool,
```
Comment (h-a-n-a) [inline crates/rspack_binding_options/src/options/raw_builtins/raw_swc_js_minimizer.rs:33]: Let's at least align these name with swc mangler, `keep_class_names`, `keep_fn_names`

File: packages/rspack/tests/configCases/builtins/minify-keep-classnames/main.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const fs = require("fs");
+const path = require("path");
+
+it("should keep class names", () => {
+	const content = fs.readFileSync(path.resolve(__dirname, "entry.js"), "utf-8");
+	expect(content).toContain("class KeepClass");
```
Comment (h-a-n-a) [inline packages/rspack/tests/configCases/builtins/minify-keep-classnames/main.js:6]: Here we can use the name field of the class

File: packages/rspack/tests/configCases/builtins/minify-keep-fn-names/main.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const fs = require("fs");
+const path = require("path");
+
+it("should keep fn names", () => {
+	const content = fs.readFileSync(path.resolve(__dirname, "entry.js"), "utf-8");
+	expect(content).toContain("function main");
```
Comment (h-a-n-a) [inline packages/rspack/tests/configCases/builtins/minify-keep-fn-names/main.js:6]: Same here

File: packages/rspack/tests/configCases/builtins/minify-keep-classnames/main.js (hunk 2 above)
Comment (xinxinhe1810) [reply packages/rspack/tests/configCases/builtins/minify-keep-classnames/main.js]: fixed

File: crates/rspack_binding_options/src/options/raw_builtins/raw_swc_js_minimizer.rs (hunk 1 above)
Comment (xinxinhe1810) [reply crates/rspack_binding_options/src/options/raw_builtins/raw_swc_js_minimizer.rs]: fixed

File: packages/rspack/tests/configCases/builtins/minify-keep-classnames/index.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+class KeepClass {
+	constructor() {
+		this.name = "test-keep-class-names";
+	}
+
+	getName() {
+		return this.name;
+	}
+}
```
Comment (h-a-n-a) [inline packages/rspack/tests/configCases/builtins/minify-keep-classnames/index.js:9]: What are these for, seems not necessary to me

**Comment by h-a-n-a** [comment]
Thank you for the help! I will merge this as soon as CI passes.

**Review by h-a-n-a — APPROVED** [review]

File: crates/rspack_testing/test.config.scheme.json [hunk 5]
```diff
@@ -346,6 +346,14 @@
           "default": false,
           "type": "boolean"
         },
+        "keepClassnames": {
```
Comment (h-a-n-a) [inline crates/rspack_testing/test.config.scheme.json:349]:
```suggestion
        "keepClassNames": {
```

File: crates/rspack_testing/test.config.scheme.json [hunk 6]
```diff
@@ -346,6 +346,14 @@
...
+          "default": false,
+          "type": "boolean"
+        },
+        "keepFnames": {
```
Comment (h-a-n-a) [inline crates/rspack_testing/test.config.scheme.json:353]:
```suggestion
        "keepFnNames": {
```

**Comment by h-a-n-a** [comment]
Would you please update the snapshot?

File: crates/rspack_plugin_swc_js_minimizer/src/lib.rs [hunk 7]
```diff
@@ -115,6 +117,14 @@ impl Plugin for SwcJsMinimizerPlugin {
       ..Default::default()
     };
 
+    println!("minify_options: {:?}", minify_options);
```
Comment (h-a-n-a) [inline crates/rspack_plugin_swc_js_minimizer/src/lib.rs:120]:
```suggestion
```

**Comment by h-a-n-a** [comment]
Let me help with the final touches

**Review by h-a-n-a — APPROVED** [review]
Thank you for the contribution!

**Review by h-a-n-a — APPROVED** [review]

**Pull request merged**